projects/prison-identifier-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/entity/ProbationSearchRepository.kt

```diff
@@ -108,12 +108,14 @@ interface ProbationSearchRepository : JpaRepository<Person, Long> {
             from Person p left join Alias a on a.offenderId = p.id
             where :croNumber is not null and lower(trim(p.croNumber)) = lower(trim(:croNumber))
             and (
-                (:surname is not null 
-                    or lower(trim(p.surname)) = lower(trim(:surname)) 
-                    or lower(trim(a.surname)) = lower(trim(:surname)))
-                or (:dateOfBirth is not null 
-                    or p.dateOfBirth = :dateOfBirth
-                    or a.dateOfBirth = :dateOfBirth)   
+                (:surname is not null and (
+                    lower(trim(p.surname)) = lower(trim(:surname)) 
+                    or lower(trim(a.surname)) = lower(trim(:surname))
+                ))
+                or (:dateOfBirth is not null and ( 
+                    p.dateOfBirth = :dateOfBirth
+                    or a.dateOfBirth = :dateOfBirth)
+                )   
```

Comment:
I might be misreading this, but do you just need to change it from `is not null` to `is null` - so skip the check if a surname filter isn't provided?
```suggestion
                (:surname is null 
                    or lower(trim(p.surname)) = lower(trim(:surname)) 
                    or lower(trim(a.surname)) = lower(trim(:surname)))
                or (:dateOfBirth is null 
                    or p.dateOfBirth = :dateOfBirth
                    or a.dateOfBirth = :dateOfBirth)  
```

Reply:
I was wrestling with this for a while and was originally doing as you suggested but it was causing issues. Imagine if they did not provide a surname, but they did provide a date of birth. It would resolve to true in the surname check and would then completely ignore the date of birth check. So checking first that its not null before doing the or on the alias and offender, then it will still do the DOB check if the surname is null